docs/developer/architecture.rst

```diff
@@ -238,39 +142,136 @@ This abstraction separates the actual data movement and function application fro
+    - More for execution flow see on :doc:`pandas on Python </flow/modin/core/execution/python/implementations/pandas_on_python/index>` page.
+- :doc:`OmniSci on Native </UsingOmnisci/index>` (experimental)
     - Uses OmniSciDB as an engine.
-    - The compute kernel/in-memory format is a pyarrow Table or pandas DataFrame when defaulting to pandas.
+    - The storage format is `omnisci` and in-memory partition type is a pyarrow Table or pandas DataFrame when defaulting to pandas.
+    - More for execution flow see on :doc:`OmniSci on Native </flow/modin/experimental/core/execution/native/implementations/omnisci_on_native/index>` page.
 - :doc:`Pyarrow on Ray </UsingPyarrowonRay/index>` (experimental)
     - Uses the Ray_ execution framework.
-    - The compute kernel/in-memory format is a pyarrow Table.
+    - The storage format is `pyarrow` and in-memory partition type a pyarrow Table.
+    - More for execution flow see on :doc:`Pyarrow on Ray </flow/modin/experimental/core/execution/ray/implementations/pyarrow_on_ray>` page.
+- cuDF on Ray (experimental)
+    - Uses the Ray_ execution framework.
+    - The storage format is `cudf` and in-memory partition type is a cuDF DataFrame.
+    - More for execution flow see on :doc:`cuDF on Ray </flow/modin/core/execution/ray/implementations/cudf_on_ray/index>` page.
 
 .. _directory-tree:
 
+DataFrame Partitioning
+----------------------
+
+The Modin DataFrame architecture follows in the footsteps of modern architectures for
+database and high performance matrix systems. We chose a partitioning schema that
+partitions along both columns and rows because it gives Modin flexibility and
+scalability in both the number of columns and the number of rows supported. The
+following figure illustrates this concept.
+
+.. image:: /img/block_partitions_diagram.png
+   :align: center
+
+Currently, main in-memory format of each partition is a `pandas DataFrame`_ (:doc:`pandas storage format </flow/modin/core/storage_formats/pandas/index>`).
+:doc:`Omnisci </flow/modin/experimental/core/storage_formats/omnisci/index>`, :doc:`PyArrow </flow/modin/core/storage_formats/pyarrow/index>`
+and cuDF are also supported as experimental in-memory formats in Modin.
+
+
+Index
+"""""
+
+We currently use the ``pandas.Index`` object for both indexing columns and rows. In the
+future, we will implement a distributed, pandas-compatible Index object in order remove
+this scaling limitation from the system. It does not start to become a problem until you
+are operating on more than 10's of billions of columns or rows, so most workloads will
+not be affected by this scalability limit. **Important note**: If you are using the
+default index (``pandas.RangeIndex``) there is a fixed memory overhead (~200 bytes) and
+there will be no scalability issues with the index.
+
+API
+"""
+
+The API is the outer-most layer that faces users. The majority of our current effort is
```

Comment:
```suggestion
The API is the outer-most layer that faces users.
```
Can we delete everything in this paragraph but this sentence? Now that we have 90%+ coverage of the pandas API, it doesn't make sense anymore.

Reply:
I think yes.